File: files/300-399/367/CH-367.py
import pandas as pd
import re

path = "300-399/367/CH-367 Text Cleaning.xlsx"
input = pd.read_excel(path, usecols="B", nrows = 7, skiprows = 2)
test = pd.read_excel(path, usecols="E", nrows=7, skiprows=2).rename(columns=lambda x: x.replace(".1", ""))

input['ID'] = input['ID'].apply(lambda x: re.sub(r'(.+)\1+', r'\1', x))

print(input.equals(test))
# True

File: files/300-399/367/CH-367.R
library(tidyverse)
library(readxl)

path <- "300-399/367/CH-367 Text Cleaning.xlsx"
input <- read_excel(path, range = "B3:B9")
test <- read_excel(path, range = "E3:E9")

result <- input %>%
  mutate(ID = str_replace(ID, "(.+)\\1+", "\\1"))

all.equal(result$ID, test$ID)
# [1] TRUE
